torch/_inductor/lowering.py

```diff
@@ -4610,6 +4610,12 @@ def reduce_scatter_tensor_coalesced(self, reduceOp, tag, ranks, group_size):
         )
         return list(map(TensorBox.create, result))
 
+    @register_lowering(c10d_functional.all_to_all_single)
+    def all_to_all_single(self, output_split_sizes, input_split_sizes, tag, ranks, group_size):
```

Comment:
I thought @eellison had a trick for getting the meta-inferred output size here.  I guess it didn't pan out for this case but I was still curious to learn about it regardless.  And is the reason that you didn't go that route because you need the symbolic expression not the constant?

Reply:
Elias taught me how to get the meta-inferred output size - I used it within the `ir.AllToAllSingle.create` method: `output_split_sizes_sum_s = V.graph.current_node.meta['val'].size()[0]`

Reply:
actually no more need to pass down meta-inferred output size after switching to the new design using ExternKernelAlloc (which also makes it much cleaner)